api/placer.py

```diff
@@ -82,13 +82,15 @@ def save_file(self, field, info):
         """
 
         # Save file
-        files.move_form_file_field_into_cas(field)
+        if field is not None:
+            files.move_form_file_field_into_cas(field)
 
         # Update the DB
-        hierarchy.upsert_fileinfo(self.container_type, self.id, info)
+        if info is not None:
+            hierarchy.upsert_fileinfo(self.container_type, self.id, info)
 
-        # Queue any jobs as a result of this upload
-        rules.create_jobs(config.db, self.container, self.container_type, info)
+            # Queue any jobs as a result of this upload
+            rules.create_jobs(config.db, self.container, self.container_type, info)
```

Comment:
This change would mean that gears will not be scheduled to run if the file does not have info. That might work for our current default set of rules, but they can arbitrarily modified on each instance's project docs. I think this breaks how gears are scheduled.